manager-rs/dauth-service/src/tasks/task_manager.rs

```diff
@@ -33,8 +33,30 @@ async fn run(context: Arc<DauthContext>) -> Result<(), DauthError> {
         if let Err(e) = tasks::register::run_task(context.clone()).await {
```

Comment:
Now that I understand the design, I think there is still a minor issue that if one of the tasks takes a long time to complete (longer than 10s) and we miss a tick, that the default behavior of tokio is to "burst" all the missed ticks immediately. Since we don't clear the update tasks from the db until they succeed, if a task takes longer than 10s to complete you can end up with a backlog. In the current always awaiting design this is not so terrible, except in the transition from when a task goes from taking a long time to not taking long at all. At that point if a big backlog has been built up, we'll start quickly making up for all the backlogged missed ticks (doing a bunch of db reads).

It might make sense to change the [missed tick behavior](https://docs.rs/tokio/latest/tokio/time/struct.Interval.html#method.set_missed_tick_behavior) to skip or delay to avoid this issue.

Reply:
Oh I see! I had thought that a tick would start a timer for 10 seconds regardless (which I didn't think would be a big deal anyway). I didn't realize it would do that

Reply:
Gotcha! Yeah I think just changing the missed tick behavior to "delay" will do what you expected it to be doing : )